deepbox/src/main/java/ch/cyberduck/core/deepbox/DeepboxIdProvider.java

```diff
@@ -0,0 +1,348 @@
+                return null;
+            }
+            else { // second+ level under Documents,Trash (Inbox has no hierarchy)
+                final String deepBoxNodeId = this.getDeepBoxNodeId(file.getParent());
+                final String boxNodeId = this.getBoxNodeId(file.getParent());
+                final String parentNodeId = this.getFileId(file.getParent());
+                if(containerService.isInDocuments(file)) {
+                    return this.lookupFileInDocumentsNodeId(file, deepBoxNodeId, boxNodeId, parentNodeId);
+                }
+                else if(containerService.isInTrash(file)) {
+                    return this.lookupFileInTrashNodeId(file, deepBoxNodeId, boxNodeId, parentNodeId);
+                }
+                return null;
+            }
+        }
+        catch(ApiException e) {
+            throw new DeepboxExceptionMappingService(this).map("Failure to read attributes of {0}", e, file);
+        }
+    }
+
+    private String lookupDeepboxNodeId(final Path file) throws ApiException {
+        final BoxRestControllerApi rest = new BoxRestControllerApi(session.getClient());
+        int size;
+        int offset = 0;
+        do {
+            final DeepBoxes deepBoxes = rest.listDeepBoxes(offset, chunksize, "displayName asc", null);
+            final String deepBoxName = file.getName();
+            final String deepBoxNodeId = deepBoxes.getDeepBoxes().stream().filter(db -> DeepboxPathNormalizer.name(db.getName()).equals(deepBoxName)).findFirst().map(db -> db.getDeepBoxNodeId().toString()).orElse(null);
+            if(deepBoxNodeId != null) {
+                this.cache(file, deepBoxNodeId);
+                return deepBoxNodeId;
+            }
+            size = deepBoxes.getSize();
+            offset += chunksize;
+        }
+        while(offset < size);
+        return this.cache(file, null);
+    }
+
+    private String lookupBoxNodeId(final Path file, final String deepBoxNodeId) throws ApiException {
+        final BoxRestControllerApi rest = new BoxRestControllerApi(session.getClient());
+        int size;
+        int offset = 0;
+        do {
+            final Boxes boxes = rest.listBoxes(UUID.fromString(deepBoxNodeId), offset, chunksize, "displayName asc", null);
+            final String boxName = file.getName();
+            final String boxNodeId = boxes.getBoxes().stream().filter(b -> DeepboxPathNormalizer.name(b.getName()).equals(boxName)).findFirst().map(b -> b.getBoxNodeId().toString()).orElse(null);
+            if(boxNodeId != null) {
+                this.cache(file, boxNodeId);
+                return boxNodeId;
+            }
+            size = boxes.getSize();
+            offset += chunksize;
+        }
+        while(offset < size);
+        return this.cache(file, null);
+    }
+
+    // N.B. we can get node id of documents - however, we might not get its nodeinfo or do listfiles from the documents root node, even if boxPolicy.isCanListFilesRoot()==true!
```

Comment:
```suggestion
    // N.B. we can get node id of documents - however, in some cases, we might not get its nodeinfo or do listfiles from the documents root node, even if boxPolicy.isCanListFilesRoot()==true! In such cases, it may be possible to delete a file (aka. move to trash) but be unable to list/find the file in the trash afterwards.
    See {@link DeepboxListServiceTest#testNoListChildrenTrashInbox}
```